gossip/src/contact_info.rs

```diff
@@ -443,6 +444,24 @@ impl ContactInfo {
     pub(crate) fn check_duplicate(&self, other: &ContactInfo) -> bool {
         self.pubkey == other.pubkey && self.outset < other.outset
     }
+
+    // Returns None if the contact-infos have different pubkey.
+    // Otherwise returns true if (self.outset, self.wallclock) tuple is larger
+    // than (other.outset, other.wallclock).
+    // If the tuples are equal it returns None.
+    #[inline]
+    #[must_use]
+    pub(crate) fn overrides(&self, other: &ContactInfo) -> Option<bool> {
+        if self.pubkey != other.pubkey {
+            return None;
+        }
```

Comment:
thinking out loud here. We check pubkeys match even though we check pubkeys match in the caller of this function: https://github.com/anza-xyz/agave/blob/ae18213c19ea5335dfc75e6b6116def0f0910aff/gossip/src/crds.rs#L198. Even though `ContactInfo::overrides()` is only called by `overrides()` in `crds.rs`, we need to ensure `ContactInfo::overrides()` acts properly within the `ContactInfo` struct itself, hence the "second" check on pubkeys.

Reply:
yeah, that is right.

My thinking is that we can't control who/where calls this function and so the function better not to presume that is always the case.
Also from a conceptual definition of "override", returning `None` when contact-infos have different pubkeys seems fine to me.